### PR TITLE
feat(kanban): sort tasks with active sessions to top of lanes

### PR DESCRIPTION
**Pull request by Maxsky5**
Sorts tasks within each kanban lane so that tasks with active agent sessions (`starting` or `running`) automatically appear at the top of their lane, before tasks without active sessions. This helps users quickly identify which tasks are being actively worked on, especially on boards with many tasks.

### Changes

`apps/desktop/src/pages/kanban/use-kanban-board-model.ts`
- Added `sortTasksByActiveSession` helper that uses `Map.has()` for O(1) active-session lookups
- Applied stable sort in `useKanbanBoardModel` hook via `columnsWithSortedTasks` memo
- Tasks with active sessions sort above inactive ones; relative order within groups is preserved

`apps/desktop/src/pages/kanban/use-kanban-board-model.test.ts`
- Added `createTaskCard` test factory
- Added 4 unit tests covering active/inactive sorting, stable sort preservation, and edge cases (empty columns, no active sessions)

### Verification

- All 6 tests pass
- `bun run typecheck` passes
- `bun run lint` passes

### Non-changes

- No changes to `KanbanColumn` or `KanbanTaskCard` rendering logic
- No changes to `buildActiveSessionsByTaskId` (already existed)
- Visual treatment of task cards remains unchanged

## Summary by CodeRabbit

* **Enhancements**
  * Kanban board columns now prioritize tasks with active sessions, showing active tasks first and inactive tasks after, while preserving original ordering within each group.
* **Tests**
  * Added tests validating active-session sorting behavior, stability of ordering, handling of empty columns, and preserving order when no active sessions exist.

### CODE REVIEW

File: apps/desktop/src/pages/kanban/use-kanban-board-model.ts [hunk 1]
```diff
@@ -44,6 +44,14 @@ export const buildRunStateByTaskId = (runs: RunSummary[]): Map<string, RunSummar
   );
 };
 
+export const sortTasksByActiveSession = (
+  tasks: TaskCard[],
+  activeSessionsByTaskId: Map<string, AgentSessionState[]>,
+): TaskCard[] => {
+  const hasActive = (task: TaskCard) => activeSessionsByTaskId.has(task.id);
+  return [...tasks].sort((a, b) => Number(hasActive(b)) - Number(hasActive(a)));
+};
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/use-kanban-board-model.ts:66]: Adopted. Replaced O(N log N) sort with O(N) single-pass partition for better performance.

File: apps/desktop/src/pages/kanban/use-kanban-board-model.ts [hunk 2]
```diff
@@ -107,10 +115,19 @@ export function useKanbanBoardModel({
 
   const activeSessionsByTaskId = useMemo(() => buildActiveSessionsByTaskId(sessions), [sessions]);
 
+  const columnsWithSortedTasks = useMemo(
+    () =>
+      columns.map((col) => ({
+        ...col,
+        tasks: sortTasksByActiveSession(col.tasks, activeSessionsByTaskId),
+      })),
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/use-kanban-board-model.ts:136]: Acknowledged as a known limitation. Fixing this properly would require pre-loading active sessions on app initialization, which is a larger architectural change beyond the scope of this PR. The sorting will work correctly once sessions are hydrated for each task.

File: apps/desktop/src/pages/kanban/use-kanban-board-model.ts [hunk 3]
```diff
@@ -107,10 +115,19 @@ export function useKanbanBoardModel({
...
+    [columns, activeSessionsByTaskId],
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply apps/desktop/src/pages/kanban/use-kanban-board-model.ts:137]: Fixed by adopting the O(N) partition approach. The partition is stable and only runs when tasks change active/inactive state, avoiding unnecessary re-computation during streaming updates.
